include/ginkgo/core/log/profiler_hook.hpp

```diff
@@ -450,7 +453,7 @@ class profiling_scope_guard {
 
     profiling_scope_guard& operator=(const profiling_scope_guard&) = delete;
 
-    profiling_scope_guard& operator=(profiling_scope_guard&&) = delete;
```

Comment:
actually I don't think we need it at all, this was from a previous implementation where I had a plain `profiling_scope_guard` lying around instead of returning from two branches of `annotate`